website/moneybirdsynchronization/signals.py

```diff
@@ -0,0 +1,359 @@
+"""The signals checked by the moneybrid synchronization package."""
+from django.db.models.signals import pre_save, post_save, pre_delete
+from django.template.defaultfilters import date
+import logging
+
+from utils.models.signals import suspendingreceiver
+
+from members.models import Member
+from moneybirdsynchronization.models import Contact
+from moneybirdsynchronization.administration import HttpsAdministration, Administration
+from moneybirdsynchronization import services
+from payments.models import Payment
+from sales.models.order import Order
+from events.models.event import Event
+from thaliawebsite import settings
+
+@suspendingreceiver(
+    post_save,
+    sender="members.Profile",
+)
+def post_profile_save(sender, instance, **kwargs):
+    """Update contact on profile creation."""
+    api = HttpsAdministration(settings.MONEYBIRD_API_KEY, settings.MONEYBIRD_ADMINISTRATION_ID)
+    member = Member.objects.get(profile=instance)
+    contact = Contact.objects.get_or_create(member=member)[0]
+    if contact.moneybird_version is None:
+        response = api.post("contacts", contact.to_moneybird())
+        contact.moneybird_id = response["id"]
+        contact.moneybird_version = response["version"]
+        contact.save()
+    else:
+        response = api.patch("contacts/{}".format(contact.moneybird_id), contact.to_moneybird())
+        contact.moneybird_version = response["version"]
+        contact.save()
+
+
+@suspendingreceiver(
+    pre_delete,
+    sender="members.Profile",
+)
+def post_profile_delete(sender, instance, **kwargs):
+    """Delete contact on profile deletion."""
+    api = HttpsAdministration(settings.MONEYBIRD_API_KEY, settings.MONEYBIRD_ADMINISTRATION_ID)
+    member = Member.objects.get(profile=instance)
+    contact = Contact.objects.get(member=member)
+    api.delete("contacts/{}".format(contact.moneybird_id))
```

Comment:
What will happen if we're deleting a person that has invoices attached to it on Moneybird? Will it delete the contact? Will it throw an error and crash this code?

Reply:
probably the latter. We could choose to not remove them if they have an invoice, but that will clutter Moneybird over time

Reply:
Moneybird archives the contact when it cannot be deleted